dot/rpc/modules/grandpa.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package modules
+
+import (
+	"net/http"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	log "github.com/ChainSafe/log15"
+)
+
+// GrandpaModule init parameters
+type GrandpaModule struct {
+	logger   log.Logger
+	blockAPI BlockAPI
+}
+
+// NewGrandpaModule creates a new Grandpa rpc module.
+func NewGrandpaModule(api BlockAPI) *GrandpaModule {
+	return &GrandpaModule{
+		blockAPI: api,
+	}
+}
+
+// ProveFinalityRequest request struct
+type ProveFinalityRequest struct {
+	blockHashStart common.Hash
+	blockHashEnd   common.Hash
+	authorityID    uint64
+}
+
+// ProveFinalityResponse is an optional SCALE encoded proof array
+type ProveFinalityResponse [][]byte
+
+// ProveFinality for the provided block range. Returns NULL if there are no known finalized blocks in the range. If no authorities set is provided, the current one will be attempted.
+func (gm *GrandpaModule) ProveFinality(r *http.Request, req *ProveFinalityRequest, res *ProveFinalityResponse) error {
+	blocksToCheck, err := gm.blockAPI.SubChain(req.blockHashStart, req.blockHashEnd)
```

Comment:
What about authorityID? I'm not sure I fully understand what it's used for, but it should be handled when the user passes in a value for this.  I think this is an ID that represents a set of authorities that were used when the given blocks were being proved (justified) so that the user that confirm that the correct authorities for that time were used. But I'm not sure?

Reply:
I think we can probably ignore it for now and open an issue for it

Reply:
Sweet I've created a ticket for it, I wasn't sure how to consume the authority id so maybe when we're a bit more clear we can update this ticket: https://github.com/ChainSafe/gossamer/issues/1404